Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -0,0 +1,361 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AnalyzerPluginGeneratorTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Plugins.Test.Common;
+using System.IO;
+using NuGet;
+using System.Linq;
+using System.Collections.Generic;
+namespace SonarQube.Plugins.Roslyn.RoslynPluginGeneratorTests
+{
+    /// <summary>
+    /// Tests for NuGetPackageHandler.cs
+    /// </summary>
+    [TestClass]
+    public class AnalyzerPluginGeneratorTests
+    {
+        public TestContext TestContext { get; set; }
+
+        private enum License { Required, NotRequired };
+        private enum Node { A, B, C, D, E, F };
+
+        [TestMethod]
+        public void Generate_PackageNoAccept_NoDependencies_Succeeds()
+        {
+            // Arrange
+            TestLogger logger = new TestLogger();
+            string outputDir = TestUtils.CreateTestDirectory(this.TestContext, ".out");
+            
+            string localNuGetSourceDir = TestUtils.CreateTestDirectory(this.TestContext, ".localNuGetSource");
+            string localNuGetDownloadDir = TestUtils.CreateTestDirectory(this.TestContext, ".localNuGetDownload");
+            IPackageManager localNuGetSourceStore = CreatePackageManager(localNuGetSourceDir);
+            SetupTestGraph(localNuGetSourceStore);
```

Comment:
This setup code is virtually identical for every test; the only difference seems to be the nodes which are marked as requiring licenses, but this difference is lost among the rest of the boilerplate code. The problem is the rest of the test needs to reference several of the variables that are assigned during set up. There are a number of possible solutions including
* creating a setup helper method that returns the required variables as output parameters (ugly)
* create a setup helper method that returns a data class that has the required data as fields (less ugly)